improver/calibration/ensemble_calibration.py

```diff
@@ -638,28 +761,76 @@ def _create_temporal_coordinates(historic_forecasts):
 
-    @staticmethod
-    def _create_spatial_coordinates(historic_forecasts):
-        """Create spatial coordinates for the EMOS coefficients cube.
-
+    def _create_spatial_dim_coordinates(self, historic_forecasts):
+        """Create spatial dimension coordinates for the EMOS coefficients cube.
         Args:
             historic_forecasts (iris.cube.Cube):
                 Historic forecasts from the training dataset.
-
         Returns:
-            list of tuples:
+            List[Tuple[iris.coords.Coord, int]]:
                 List of tuples of the spatial coordinates and the associated
                 dimension. This format is suitable for use by iris.cube.Cube.
         """
-        spatial_coords_and_dims = []
+        if not any([self.each_point, self.minimise_each_point]):
+            return []
+
+        spatial_coord_dims = []
         for axis in ["x", "y"]:
-            spatial_coords_and_dims.append(
-                (historic_forecasts.coord(axis=axis).collapsed(), None)
+            spatial_coord_dims.append(
+                historic_forecasts.coord_dims(
+                    historic_forecasts.coord(axis=axis).name()
+                )
+            )
+        dim_coords_and_dims = []
+        # Remove dims where x and y axis share a dimension coordinate (as for a spot forecast cube)
+        spatial_coord_dims = list(set(spatial_coord_dims))
+        for index, dim in enumerate(spatial_coord_dims):
+            (spatial_dim_coord,) = historic_forecasts.coords(
+                dimensions=dim, dim_coords=True
             )
-        return spatial_coords_and_dims
+            dim_coords_and_dims.append((spatial_dim_coord, index))
+        return dim_coords_and_dims
+
+    def _create_spatial_aux_coordinates(self, historic_forecasts):
+        """Create spatial auxiliary coordinates for the EMOS coefficients cube.
+        Args:
+            historic_forecasts (iris.cube.Cube):
+                Historic forecasts from the training dataset.
+        Returns:
+            List[Tuple[iris.coords.Coord, int or None]]:
+                List of tuples of the spatial coordinates and the associated
+                dimension. This format is suitable for use by iris.cube.Cube.
+        """
+        if self.each_point | self.minimise_each_point:
+            aux_coords_and_dims = []
+            for index, axis in enumerate(["x", "y"]):
+                spatial_coord_dim = historic_forecasts.coord_dims(
+                    historic_forecasts.coord(axis=axis).name()
+                )
+                for coord in historic_forecasts.coords(
+                    dimensions=spatial_coord_dim, dim_coords=False
+                ):
+                    if not any(
+                        [True if c[0] == coord else False for c in aux_coords_and_dims]
+                    ):
+                        aux_coords_and_dims.append(
+                            (historic_forecasts.coord(coord), index)
+                        )
```

Comment:
I've updated this as suggested. I've also made a few other edits in this area of the code for creating the coefficient cubes, particularly related to the `realization` coordinate that can exist on the `beta` coefficient cube, if the `realizations` are used as the predictor, rather than the ensemble mean.